app/src/main/java/org/mozilla/focus/ext/When.kt

```diff
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.ext
+
+/**
+ * `when` is only exhaustive when its expression returns a value. Appending this
+ * value gives it a return value, forcing the compiler to check `when` for
+ * exhaustiveness.
+ *
+ * One downside is that it pollutes autocomplete, but let's try it out.
+ *
+ * Idea from SO: https://stackoverflow.com/a/44383076
+ */
+val Any?.forceExhaustive get() = Unit
```

Comment:
Okay, so for void return types for when, it doesn't need to be exhaustive? That seems okay. How do you enforce this?

Reply:
Yes, for `Unit` return types (void doesn't exist in Kotlin).

You can't enforce this, but if someone remembers to use it, the compiler won't let you compile without it. The alternatives weren't as easy to understand so I went with this one but I'm seeing more problems:
- It is literally first in autocomplete always
- Compiler warns about the return types of my functions (in my next PR)

I thought of one more alternative not addressed on SO: have a global `whenExhaustive` function. We can rethink this if it becomes a problem.